packages/visual-editing/src/controller.ts

```diff
@@ -1,20 +1,27 @@
 import {v4 as uuid} from 'uuid'
 
 import type {
+  DragState,
   ElementNode,
   EventHandlers,
   OverlayController,
   OverlayElement,
   OverlayOptions,
   ResolvedElement,
 } from './types'
-import {findSanityNodes} from './util/findSanityNodes'
+import {handleOverlayDrag} from './util/dragAndDrop'
+import {findSanityNodes, isSanityNode, sanityNodesExistInSameArray} from './util/findSanityNodes'
 import {getRect} from './util/getRect'
 
 const isElementNode = (target: EventTarget | null): target is ElementNode => {
   return target instanceof HTMLElement || target instanceof SVGElement
 }
 
+const dragState: DragState = {
+  status: 'idle',
+  insertPosition: null,
+}
```

Comment:
Feel like I'm missing something here but I'll ask the question anyway: What is the `dragState` useful for in this scope? Is the shared state between  `handleOverlayDrag` doing something I haven't picked up on?

Also, what is the `status` value used for? I can see that it's mutated both in this file and within the utility function, but to what end?

Reply:
Good question! Currently, it tracks `insertPosition` so that we only update/emit events when a new position in calculated. 

`status` will be set to `dragging` when we start the UI work on `mousedown` so that we can disable certain overlay behaviour during a drag sequence :)

Reply:
Would it make sense to add something like `dragStatus` to the `OverlayState` interface so we have a single place for storing overlay related state, and then track the previous `insertPosition` within the `handleOverlayDrag` function (that is if that doesn't need to be shared state?)

Reply:
That makes total sense. I have gone ahead and remove `dragState` here and added a simple `prevInsertPosition` to the `handleOverlayDrag` function. We can add the UI-specific events next if that works, and integrate with the `overlayState` reducer :)